src/java/org/apache/cassandra/index/sai/StorageAttachedIndex.java

```diff
@@ -390,10 +390,15 @@ public Callable<?> getPreJoinTask(boolean hadBootstrap)
     public Callable<?> getTruncateTask(long truncatedAt)
     {
         /*
-         * index files will be removed as part of base sstable lifecycle in
-         * {@link LogTransaction#delete(java.io.File)} asynchronously.
+         * index files will be removed as part of base sstable lifecycle in {@link LogTransaction#delete(java.io.File)}
+         * asynchronously, but we need to mark the index queryable because if the truncation is during the initial
+         * build of the index it won't get marked queryable by the build.
          */
-        return null;
+        return () -> {
+            logger.info(indexIdentifier.logMessage("Making index queryable during table truncation"));
```

Comment:
```suggestion
            logger.info(indexIdentifier.logMessage("Making index queryable during table truncation..."));
```
nit: Totally don't have to do this, but I like throwing ellipses on there when it's something we're about to do ;)

Reply:
I don't have an issue with that, except that I'd usually leave it for something that was going to take a little time to complete. This is just setting a flag. If you feel it needs it I'm happy for the change.

Reply:
Yeah, that makes sense. The other thing I forgot is that `makeIndexQueryable()` has its own logging, maybe adding a new `Index.Status` for truncation would allow us to just piggyback on that...

```
logger.info("Index [{}] became queryable after successful build.", name);
```

...message?

Reply:
It starts to get messy if we do that. There is a lot of code that depends on `BUILD_SUCCEEDED`. If we add a new status, we would need to handle that correctly. I'd prefer to leave this as is. As you say, truncation is a bad idea anyway.